marine.c

```diff
@@ -633,7 +634,50 @@ WS_DLL_PUBLIC int validate_display_filter(char *dfilter) {
     return TRUE;
 }
 
-WS_DLL_PUBLIC int marine_add_filter(char *bpf, char *dfilter, char **fields, int fields_len, char *err_msg) {
+int parse_output_fields(output_fields_t *output_fields, char **fields, int fields_len, char **err_msg) {
+    for (int i = 0; i < fields_len; i++) {
+        output_fields_add(output_fields, fields[i]);
+    }
+
+    GSList *it = NULL;
+    GSList *invalid_fields = output_fields_valid(output_fields);
+    if (invalid_fields != NULL) {
+        if (err_msg == NULL) {
+            return -1;
+        }
+
+        int current_err_size = 64;
+        *err_msg = (char *)malloc(current_err_size);
+        strcpy(*err_msg, "Some fields aren't valid:\n");
+
+        for (it = invalid_fields; it != NULL; it = g_slist_next(it)) {
+            current_err_size = current_err_size + strlen((gchar *) it->data) + 1;
```

Comment:
don't you need to add 2?
you adding one for the 1 for the null terminanator, and you adding that `\t` aftrer so it should be 2?

Reply:
I don't need to add a null terminator after each field as we only need one for the entire message which comes when we copy the initial message.